sys/include/net/gnrc/dns/internal.h

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (C) 2015 Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    net_dns DNS
+ * @ingroup     net
+ * @brief       Provides DNS header and helper functions
+ * @{
+ *
+ * @file
+ * @brief       Definitions for DNS
+ *
+ * @author      Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ */
+
+#ifndef GNRC_DNS_INTERNALS_H_
+#define GNRC_DNS_INTERNALS_H_
+
+#include <errno.h>
+#include <string.h>
+#include <time.h>
+
+#include "net/gnrc/dns.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/netapi.h"
+#include "net/gnrc/netreg.h"
+#include "net/gnrc/pktbuf.h"
+#include "net/gnrc/udp.h"
+#include "xtimer.h"
+
+/**
+ * @brief   Enable/Disable debug trace support
+ */
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Thread ID of the DNS server
+ */
+extern kernel_pid_t _dns_pid;
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+/**
+ * @brief   Compile time byte-order swapping for short constants
+ */
+#define CT_HTONS(x)         ((uint16_t)(          \
+                                 ((uint16_t)x >> 8) & 0x00FF) | \
+                             (((uint16_t)x << 8) & 0xFF00))
+
+/**
+ * @brief   Compile time byte-order swapping for integers constants
+ */
+#define CT_HTONL(x)         ((uint32_t)(          \
+                                 ((uint32_t)x >> 24) & 0x000000FF) | \
+                             (((uint32_t)x >> 8) & 0x0000FF00) | \
+                             (((uint32_t)x << 8) & 0x00FF0000) | \
+                             (((uint32_t)x << 24) & 0xFF000000))
+#else
+/**
+ * @brief   Compile time byte-order swapping for short constants
+ */
+#define CT_HTONS(x)    ((uint16_t)x)
+
+/**
+ * @brief   Compile time byte-order swapping for integers constants
+ */
+#define CT_HTONL(x)    ((uint32_t)x)
+#endif
```

Comment:
Shouldn't these defines be removed here and make the PR dependent on #4812 instead?